app/src/main/java/org/schabi/newpipe/player/playback/TrackSelectionListener.java

```diff
@@ -0,0 +1,122 @@
+        // The auto string
+        final String qualityTextViewText = qualityTextView.getText().toString();
+
+        final int itemId = item.getItemId();
+        if (itemId == 0) {
+            // 0 is the index of the Auto item of the quality selector
+            trackSelector.setParameters(builder);
+
+            // If the quality text view text contains already the auto string, it means the user
+            // already selected the auto option and pressed again the auto option from the quality
+            // selector, so no need to update the text of the quality text view (if we update it,
+            // we will get something like Auto (Auto (1080p))).
+            if (!qualityTextViewText.contains(context.getString(R.string.auto_quality))) {
+                qualityTextView.setText(context.getString(R.string.auto_quality_selected,
+                        context.getString(R.string.auto_quality),
+                        qualityTextViewText));
+            }
+        } else {
+            // A resolution string
+            TrackGroup videoTrackGroup = null;
+            int lengthOfVideoTrackGroupWhichMatchMenuItem = 0;
+            for (int i = 0; i < rendererTrackGroups.length; i++) {
+                final TrackGroup trackGroup = rendererTrackGroups.get(i);
+                final int rendererTrackGroupLength = trackGroup.length;
+                if (rendererTrackGroupLength >= itemId) {
+                    lengthOfVideoTrackGroupWhichMatchMenuItem = rendererTrackGroupLength;
+                    videoTrackGroup = trackGroup;
+                }
+            }
+
+            final int qualityId = lengthOfVideoTrackGroupWhichMatchMenuItem > 0
+                    ? lengthOfVideoTrackGroupWhichMatchMenuItem - itemId
+                    : itemId - 1;
+            final DefaultTrackSelector.SelectionOverride selectionOverride =
+                    new DefaultTrackSelector.SelectionOverride(
+                            rendererIndex, qualityId);
+            builder.setSelectionOverride(rendererIndex, rendererTrackGroups,
+                    selectionOverride);
+
+            // The quality needs only to be set here when a user keeps the quality selected
+            // automatically by ExoPlayer but selects it manually with the quality selector (the
+            // quality string will be changed in other cases with the analytics listener set in the
+            // player)
+            if (qualityTextViewText.contains(context.getString(R.string.auto_quality))) {
+                final Format currentPlayingFormat = videoTrackGroup != null
+                        ? videoTrackGroup.getFormat(qualityId)
+                        // Fallback to the first track group
+                        : rendererTrackGroups.get(0).getFormat(qualityId);
+                qualityTextView.setText(getResolutionStringFromFormat(context,
+                        currentPlayingFormat));
+            }
+
+            trackSelector.setParameters(builder);
```

Comment:
@litetex is right, this can be put at the bottom of the function just before `return true` since it is called always.